chickadee/processes/wps_rerank.py

```diff
@@ -0,0 +1,138 @@
+import os
+import re
+from pywps import Process, LiteralInput, ComplexInput, FORMATS
+from pywps.app.Common import Metadata
+from netCDF4 import Dataset
+
+from wps_tools.utils import log_handler
+from wps_tools.io import log_level, nc_output
+from chickadee.utils import (
+    logger,
+    set_end_date,
+    get_package,
+    collect_args,
+    common_status_percentage,
+)
+from chickadee.io import gcm_file, obs_file, varname, out_file, num_cores, end_date
+
+
+class Rerank(Process):
+    """Quantile Reranking fixes bias introduced by the Climate Analogues
+    step by re-applying a simple quantile mapping bias correction at
+    each grid box"""
+
+    def __init__(self):
+        self.status_percentage_steps = common_status_percentage
+
+        inputs = [
+            obs_file,
+            varname,
+            out_file,
+            num_cores,
+            log_level,
+            ComplexInput(
+                "qdm_file",
+                "QDM NetCDF file",
+                abstract="Filename of output from QDM step",
+                min_occurs=1,
+                max_occurs=1,
+                supported_formats=[FORMATS.NETCDF, FORMATS.DODS],
+            ),
+            LiteralInput(
+                "analogues_object",
+                "Analogues R object",
+                abstract="R object containing the analogues produced from the CA step (suffix .rds)",
+                min_occurs=0,
+                max_occurs=1,
+                data_type="string",
+            ),
+        ]
+
+        outputs = [nc_output]
+
+        super(Rerank, self).__init__(
+            self._handler,
+            identifier="rerank",
+            title="Rerank",
+            abstract="Quantile Reranking fixes bias introduced by the Climate Analogues step",
+            keywords=["quantile reranking", "rerank", "bias correction"],
+            metadata=[
+                Metadata("NetCDF processing"),
+                Metadata("Climate Data Operations"),
+                Metadata("PyWPS", "https://pywps.org/"),
+                Metadata("Birdhouse", "http://bird-house.github.io/"),
+                Metadata("PyWPS Demo", "https://pywps-demo.readthedocs.io/en/latest/"),
+            ],
+            version="0.1.0",
+            inputs=inputs,
+            outputs=outputs,
+            store_supported=True,
+            status_supported=True,
+        )
+
+    def _handler(self, request, response):
+        loglevel = request.inputs["loglevel"][0].data
+        log_handler(
+            self,
+            response,
+            "Starting Process",
+            logger,
+            log_level=loglevel,
+            process_step="start",
+        )
+
+        (
+            obs_file,
+            varname,
+            out_file,
+            num_cores,
```

Comment:
Does `num_cores` have a default value? I notice you are providing cores in the test file.

Reply:
It does, I just thought it would be good to test :woman_shrugging:

Reply:
Oh You're talking pytest, I was thinking demo. I guess it's not necessary for the pytest, you're right